home-assistant/android
# WIP: Add nfc support

The MR adds basic support for reading and writing of nfc tags as requested in https://github.com/home-assistant/android/issues/431.

As suggested by @dshokouhi and @GrandNewbien I implemented NFC support by adding basic [universal link](https://companion.home-assistant.io/docs/integrations/universal-links) support to the app. However due to the fact that iOS uses `www.home-assistant.io/ios/nfc/` in the universal links, they won't be interchangeable between the platforms. Maybe it would make sense to agree upon a standard so that nfc tags work on both platforms.

The UI-Code inside the `NfcActivity` is heavily influenced by the `ButtonWidgetConfigureActivity`. As I'm new to kotlin development there might be some parts of the code that probably could be optimize. Might also make sense to abstract this even more and make one generic activity that contains the logic to create the definition for service calls.

-----

How to test:

- check out the [branch](https://github.com/David-Development/android-1/tree/nfc-support)
- build and run the app
- open the app
- go to the "app configuration"
- scroll down, click on "NFC Tags" to start generating new nfc tags
